zebra-rpc/src/methods.rs

```diff
@@ -995,13 +974,16 @@ where
                 let mut nonce = *header.nonce;
                 nonce.reverse();
 
-                let final_sapling_root: [u8; 32] = if sapling_tree.position().is_some() {
-                    let mut root: [u8; 32] = sapling_tree.root().into();
-                    root.reverse();
-                    root
-                } else {
-                    [0; 32]
-                };
+                let sapling_activation = NetworkUpgrade::Sapling.activation_height(&network);
+                let sapling_tree_size = sapling_tree.count();
+                let final_sapling_root: [u8; 32] =
+                    if sapling_activation.is_some() && height >= sapling_activation.unwrap() {
```

Comment:
`sapling_activation.unwrap()` should be fine since even configured Testnets must have an activation height for Canopy and all prior network upgrades.